packages/polling-controller/src/BlockTrackerPollingController.ts

```diff
@@ -0,0 +1,97 @@
+import { BaseController, BaseControllerV1 } from '@metamask/base-controller';
+import type {
+  NetworkClientId,
+  NetworkClient,
+} from '@metamask/network-controller';
+import type { Json } from '@metamask/utils';
+
+import {
+  PollingControllerBaseMixin,
+  getKey,
+} from './PollingController-abstract';
+import type { PollingTokenSetId } from './PollingController-abstract';
+
+type Constructor = new (...args: any[]) => object;
+
+/**
+ * BlockTrackerPollingControllerMixin
+ *
+ * @param Base - The base class to mix onto.
+ * @returns The composed class.
+ */
+function BlockTrackerPollingControllerMixin<TBase extends Constructor>(
+  Base: TBase,
+) {
+  abstract class BlockTrackerPollingController extends PollingControllerBaseMixin(
+    Base,
+  ) {
+    #activeListeners: Record<string, (options: Json) => Promise<void>> = {};
+
+    constructor() {
+      super();
+      this.setPollingStrategy({
+        start: (networkClientId: NetworkClientId, options: Json) =>
+          this.startBlockTrackingPolling(networkClientId, options),
+        stop: (key: PollingTokenSetId) => this.stopBlockTrackingPolling(key),
+      });
+    }
+
+    abstract getNetworkClientById(
```

Comment:
Is this something that can be defined in the abstract PollingController? It seems like something that would be common to both.